### PR TITLE
Export v2

### PR DESCRIPTION
**Pull request by gacarrillor**
This PR brings Export back to life. 

We are done adding validation and GUI improvements!

### CODE REVIEW

File: projectgenerator/gui/export.py [hunk 1]
```diff
@@ -19,12 +19,16 @@
 """
 
 import os
+from PyQt5.QtCore import Qt
+
+from PyQt5.QtWidgets import QCompleter
```
Comment (m-kuhn) [inline projectgenerator/gui/export.py]: Can you rewrite these to the same style as below? They both do the same, just for consistency.

    from qgis.PyQt.QtWidgets import xxx

File: projectgenerator/gui/export.py [hunk 2]
```diff
@@ -46,9 +50,32 @@ def __init__(self, base_config, parent=None):
         self.base_configuration = base_config
         self.restore_configuration()
 
+        self.validators = Validators()
+        regexp = QRegExp("[a-zA-Z_0-9]+") # Non empty string
```
Comment (m-kuhn) [inline projectgenerator/gui/export.py]: I think for hostnames there are other characters allowed (or even expected) like `.` (ip addresses), `-` etc. Is this properly handled here? Maybe it's better to really just check for anything else than empty string?

Comment (gacarrillor) [inline projectgenerator/gui/export.py]: Fertig!

**Comment by m-kuhn** [comment]
Nice, this looks good.

One last question, is it required to provide it with a .ili file? I thought that an export could also get the model information from a model in a repository (or even in the database?)

**Comment by gacarrillor** [comment]
Right, we don't need an ili file for exporting data. As far as I know, ili2pg reads model definition from the database itself. However, we need to pass the 'scope' we'd like to limit data export to. ili2pg supports these scopes for exporting data, models, topics, baskets, or datasets. One of them must be given (see https://github.com/claeis/ili2db/blob/master/src/ch/ehi/ili2db/base/Ili2db.java#L1276). We are currently just using models, but we could add support for other scopes as soon as we need them.

**Comment by gacarrillor** [comment]
Right, a model would be enough. Could you please point me to what leads you to think about an ili file?

**Comment by m-kuhn** [comment]
Can't find it any more, all good :)
Nice work

**Pull request merged**